File: SocialAppFlutter/social_app/lib/services/chat/received_friend_request_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'json_generated_code/received_friend_request_model.g.dart';

@JsonSerializable(explicitToJson: true)
class ReceivedFriendRequestModel {
  final int id;
  final String firstname;
  final String lastname;
  List<int>? profileImage;
  final int friendRequestId;

  ReceivedFriendRequestModel(this.id, this.firstname, this.lastname,this.profileImage, this.friendRequestId);

  factory ReceivedFriendRequestModel.fromJson(Map<String, dynamic> json) => _$ReceivedFriendRequestModelFromJson(json);

  Map<String, dynamic> toJson() => _$ReceivedFriendRequestModelToJson(this);
}

File: APIs/ChatService/ChatService/DTOs/ReceivedFriendRequestModel.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace ChatService.DTOs
{
    public class ReceivedFriendRequestModel
    {
        [JsonProperty("id")]
        public int Id { get; set; }
        [JsonProperty("firstname")]
        public string Firstname { get; set; } = string.Empty;
        [JsonProperty("lastname")]
        public string Lastname { get; set; } = string.Empty;
        [JsonProperty("profileImage")]
        public List<int>? ProfileImage { get; set; } = null!;
        [JsonProperty("friendRequestId")]
        public int FriendRequestId;
    }
}
